packages/client/tests/functional/extensions/enabled/result.ts

```diff
@@ -206,6 +206,24 @@ testMatrix.setupTestSuite(() => {
     expect(user?.loudName).toBe('JOHN SMITH')
   })
 
+  test('shadowing dependency', async () => {
```

Comment:
Would it be useful to add a test that shadows a shadowed field like in the linked issue?

Reply:
This was a really good suggestion since the functionality was actually broken. Now it's fixed and PR is ready for another re-review.